tests/test_updater_with_simulator.py

```diff
@@ -115,13 +114,40 @@ def test_targets(self, test_case_data: Tuple[str, str]):
         with open(local_path, "rb") as f:
             self.assertEqual(f.read(), content)
 
-        if "/" in targetpath:
-            # assert local_path != targetpath because of the URL encoding
-            # make target_path absolute as local_path
-            target_path = os.path.join(self.targets_dir, targetpath)
-            self.assertNotEqual(target_path, local_path)
+    def test_targets_with_subdirectories(self):
```

Comment:
We've misunderstood each other, sorry about that. Duplicating the whole piece of code like this makes no sense. 
I imagined a couple of lines downloading a file and comparing the paths.

But let's forget about the separate test. I suggest we do something else. In the targets DataSet, the Tuple can be expanded with one more item representing the encoded targethpath. In the ascii case this will duplicate the targetpath. For example:

```python
    targets: utils.DataSet = {
        "standard case": ("targetpath", b"content", "targetpath"),
        "non-asci case": ("åäö", b"more content", "%C3%A5%C3%A4%C3%B6"),
        "subdirectory case": ("a/b/c/targetpath", b"dir target content", "a%2Fb%2Fc%2Ftargetpath"),
    }
```
You can maybe use a named-tuple if you think this is not readable enough. 
At the end of `test_targets`, instead of asserting that the two paths are not equal, we can assert that the local path  really is the encoded string from the DataSet:
`self.assertEqual(local_path, os.path.join(self.targets_dir, encoded_path))`

What do you think about this option?

Reply:
It worked well I think.
You can have a look now.